### DISCUSSION TITLE
Microsoft Odata Client - Case insensitive

### DISCUSSION THREAD

**Discussion by Gabriel-Goldschmitt** [OData/odata.net#3133] [Q&A]
Hello Community,

I apologize if this is documented and I missed it.

Normally, with the generator, you have the correct attribution, so in this case, it doesn't matter. However, if you don't use them for some reason, this could be an issue.

A typical C# class uses Pascal case, but the default configuration for the OData Server is camel case, so the deserialization doesn't work without the attribution. I know case insensitivity has some performance disadvantages, but is it possible to configure the JSON deserialization in that way?

Thanks in advance.

**Comment by Gabriel-Goldschmitt** [comment]
Short update, I tried to add the following lines

```
this.Configurations.ResponsePipeline.OnMessageReaderSettingsCreated(x =>
{
    x.Settings.EnablePropertyNameCaseInsensitive = true;
});
```

But seems not to work the properties are still null-ish. I will create a sample project and see if this works there.

**Comment by mdsharpe** [comment]
Having the same trouble here. Cannot figure out how to make the OData client case-insensitive.

**Comment by mdsharpe** [comment]
I suspect it's possibly due to line 231 in `ClientTypeAnnotation`:

```c#
private void BuildPropertyCache()
{
    ....
    var propertyCache = new Dictionary<string, ClientPropertyAnnotation>(StringComparer.Ordinal);
```

From my adventures through the codebase this seems to be where the client attempts to find each property by name, and that hard-coded Ordinal string comparison is not respecting the `EnablePropertyNameCaseInsensitive` setting.
